ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -138,17 +153,22 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
             connId = selector.connect(new InetSocketAddress(host, port.getPort()), networkConfig.socketSendBufferBytes,
                 networkConfig.socketReceiveBufferBytes, port.getPortType());
             connectionTracker.startTrackingInitiatedConnection(host, port, connId);
+            requestMetadata.connectionCheckOutAttempts++;
```

Comment:
don't you need to update this right after the checkout? why is it being done only when a `connect` is made?

Also, is this metric really useful? You do have a metric for number of connections timing out due to unavailability of connections - is that not good enough? what purpose does knowing the number of attempts made serve - that is totally dependent on the number of times `sendAndPoll()` is called within the network client timeout.